microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -0,0 +1,263 @@
+            the required one. (For Redhat 8, no need to modify grub config file. It can
+            specify crashkernel by using grubby command directly)
+        b. Update grub config
+    3. Reboot system to make kdump effect.
+    4. Check if the crash kernel is loaded.
+        a. Check if kernel cmdline has crashkernel option and the value is expected.
+        b. Check if /sys/kernel/kexec_crash_loaded file exists and the value is '1'.
+        c. Check if /proc/iomem is reserved memory for crash kernel
+    5. Trigger kdump through 'echo c > /proc/sysrq-trigger'
+    6. Check if vmcore or dump file is generated under'/var/crash' after system boot up.
+
+    It has 7 test cases in this test suite. They verfy if the kdump is effect when:
+        1. with different vcpu count, 1, or more than 2, 16.
+        2. the crashkernel is set auto.
+        3. with a large count cpu, such as VM size E96a_v4.
+        4. with a large memory, such as M128s, whose memory is 2T.
+    """,
+)
+class KdumpCrash(TestSuite):
+    crash_kernel = "512M"
+    timeout_of_dump_crash = 60
+
+    @TestCaseMetadata(
+        description="""
+        This test case verfiy if the kdump is effect when VM has one vcpu.
+        The VM need 2G memory to make sure there are enough memory for crash kernel
+        Please refer to test suite specification to see the steps.
+        """,
+        priority=1,
+        requirement=node_requirement(
+            node=schema.NodeSpace(
+                core_count=1, memory_mb=search_space.IntRange(min=2048)
+            ),
+        ),
+    )
+    def kdumpcrash_validate_single_core(self, node: Node, log: Logger) -> None:
+        self._kdump_test(node, log)
+
+    @TestCaseMetadata(
+        description="""
+        This test case verfiy if the kdump is effect when VM has more than one vcpu.
+        Please refer to test suite specification to see the steps.
+        """,
+        priority=1,
+        requirement=simple_requirement(min_core_count=2),
+    )
+    def kdumpcrash_validate_smp(self, node: Node, log: Logger) -> None:
+        self._kdump_test(node, log)
+
+    @TestCaseMetadata(
+        description="""
+        This test case verfiy if the kdump is effect when crashkernel is set auto.
+        Please refer to test suite specification to see the steps.
+        """,
+        priority=2,
+    )
+    def kdumpcrash_validate_auto_size(self, node: Node, log: Logger) -> None:
+        self.crash_kernel = "auto"
+        self._kdump_test(node, log)
+
+    @TestCaseMetadata(
+        description="""
+        This test case verfiy if the kdump is effect when crashkernel is set auto and
+        the memory is more than 2T. With the crashkernel=auto parameter, system will
+        reserved a suitable size memory for crash kernel. We want to see if the
+        crashkernel=auto can also handle this scenario when the system memory is large.
+
+        Please refer to test suite specification to see the steps.
```

Comment:
Let's write it in a test case, not test suite. The test suite are not recorded in notifier, so it won't be in our database.